vm_core.h

```diff
@@ -863,10 +863,12 @@ typedef struct rb_control_frame_struct {
     const VALUE *ep;        // cfp[4] / block[1]
     const void *block_code; // cfp[5] / block[2] -- iseq, ifunc, or forwarded block handler
     void *jit_return;       // cfp[6] -- return address for JIT code
-    int generation;
+    int generation;         // cfp[6] new added
+    VALUE method;           // cfp[7] new added
+    VALUE ep2;              // cfp[8] new added
```

Comment:
save ep and method directly as the *sp may change.

could consider other solution, such as walk through the sp